examples_extra/TextStressTest/Sample/ViewController.m

```diff
@@ -0,0 +1,214 @@
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#import "ViewController.h"
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+#define NUMBER_ELEMENTS 5
+
+@interface ViewController ()
+{
+  NSMutableArray <ASTextNode *> *_textNodes;
+  NSMutableArray <UILabel *> *_textLabels;
+  UIScrollView *_scrollView;
+}
+
+@end
+
+
+@implementation ViewController
+
+- (instancetype)init
+{
+  if (!(self = [super init]))
+    return nil;
+  
+  _scrollView = [[UIScrollView alloc] init];
+  _textNodes = [NSMutableArray array];
+  _textLabels = [NSMutableArray array];
+
+  return self;
+}
+
+- (void)viewDidLoad
+{
+  [super viewDidLoad];
+  
+  [self.view addSubview:_scrollView];
+  
+  for (int i = 0; i < NUMBER_ELEMENTS; i++) {
+    
+    ASTextNode *node = [self createNodeForIndex:i];
+    [_textNodes addObject:node];
+    [_scrollView addSubnode:node];
+
+    UILabel *label = [self createLabelForIndex:i];
+    [_textLabels addObject:label];
+    [_scrollView addSubview:label];
+  }
+}
+
+- (void)viewWillLayoutSubviews
+{
+  [super viewWillLayoutSubviews];
+  
+  CGFloat maxWidth = 0;
+  CGFloat maxHeight = 0;
+  
+  CGRect frame = CGRectMake(50, 50, 0, 0);
+  
+  for (int i = 0; i < NUMBER_ELEMENTS; i++) {
+    
+    frame.size = [self sizeForIndex:i];
+    [[_textNodes objectAtIndex:i] setFrame:frame];
+    
+    frame.origin.x += frame.size.width + 50;
+    
+    [[_textLabels objectAtIndex:i] setFrame:frame];
+    
+    if (frame.size.width > maxWidth) {
+      maxWidth = frame.size.width;
+    }
+    if ((frame.size.height + frame.origin.y) > maxHeight) {
+      maxHeight = frame.size.height + frame.origin.y;
+    }
+    
+    frame.origin.x -= frame.size.width + 50;
+    frame.origin.y += frame.size.height + 20;
+  }
+  
+  _scrollView.frame = self.view.bounds;
+  _scrollView.contentSize = CGSizeMake(maxWidth, maxHeight);
+}
+
+- (ASTextNode *)createNodeForIndex:(NSUInteger)index
+{
+  ASTextNode *node = [[ASTextNode alloc] init];
+  node.attributedText = [self textForIndex:index];
+  node.backgroundColor = [UIColor orangeColor];
+  
+  NSMutableAttributedString *string = [node.attributedText mutableCopy];
+  
+  switch (index) {
+    case 0:                             // top justification (ASDK) vs. center justification (UILabel)
+      node.maximumNumberOfLines = 3;
+      return node;
+      
+    case 1:                             // default truncation attributed string color shouldn't match attributed text color (ASDK) vs. match (UIKit)
+      node.maximumNumberOfLines = 3;
+      [string addAttribute:NSForegroundColorAttributeName
+                     value:[UIColor redColor]
+                     range:NSMakeRange(0, [string length])];
+      node.attributedText = string;
+      return node;
+      
+    case 2:
+      node.maximumNumberOfLines = 3;
```

Comment:
Can you remove these additional cases for now?  They're easy to add back in as we implement more.